File: app/src/main/java/com/manimarank/spell4wiki/models/WikiSearchWords.java
package com.manimarank.spell4wiki.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class WikiSearchWords {

    @SerializedName("continue")
    @Expose
    private Offset offset;
    @SerializedName("query")
    @Expose
    private Query query;

    public Offset getOffset() {
        return offset;
    }

    public void setOffset(Offset offset) {
        this.offset = offset;
    }

    public Query getQuery() {
        return query;
    }

    public void setQuery(Query query) {
        this.query = query;
    }

    public static class Offset {

        @SerializedName("sroffset")
        @Expose
        private Integer nextOffset;

        public Integer getNextOffset() {
            return nextOffset;
        }

        public void setNextOffset(Integer nextOffset) {
            this.nextOffset = nextOffset;
        }
    }

    public static class Query {

        @SerializedName("search")
        @Expose
        private List<WikiWord> wikiTitleList = new ArrayList<>();

        public List<WikiWord> getWikiTitleList() {
            return wikiTitleList;
        }

        public void setWikiTitleList(List<WikiWord> wikiTitleList) {
            this.wikiTitleList = wikiTitleList;
        }
    }

    public static class WikiWord {

        @SerializedName("pageid")
        @Expose
        private Integer pageId;
        @SerializedName("title")
        @Expose
        private String title;

        public Integer getPageId() {
            return pageId;
        }

        public void setPageId(Integer pageId) {
            this.pageId = pageId;
        }

        public String getTitle() {
            return title;
        }

        public void setTitle(String title) {
            this.title = title;
        }
    }

}

File: app/src/main/java/com/manimarank/spell4wiki/data/model/WikiSearchWords.kt
package com.manimarank.spell4wiki.data.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

/**
 * Model class for Wiki Search Words
 * @property offset Offset?
 * @property query SearchQuery?
 * @constructor
 */
data class WikiSearchWords(
    @SerializedName("continue")
    var offset: Offset? = null,
    @SerializedName("query")
    var query: SearchQuery? = null
)

/**
 * Model class for Offset Object
 * @property nextOffset Int?
 * @constructor
 */
data class Offset(
    @SerializedName("sroffset")
    var nextOffset: Int? = null
)

/**
 * Model Class for Search Query
 * @property wikiTitleList List<WikiWord>
 * @constructor
 */
data class SearchQuery(
    @SerializedName("search")
    @Expose
    var wikiTitleList: List<WikiWord> = ArrayList()
)

/**
 * Model Class for Wiki Word
 * @property pageId Int?
 * @property title String?
 * @constructor
 */
data class WikiWord(
    @SerializedName("pageid")
    var pageId: Int? = null,
    @SerializedName("title")
    var title: String? = null
)
